src/mpi/pt2pt/bsendutil.c

```diff
@@ -571,7 +569,7 @@ static int bsend_isend_user(struct MPII_BsendBuffer_user *user, MPI_Aint packsiz
              * either primitive or contiguous types, and just
              * use MPIR_Memcpy and the provided datatype */
             msg->count = 0;
-            if (dtype != MPI_PACKED) {
+            if (1) {
```

Comment:
```suggestion
```
Suggest to remove this branch and related dead code in the else clause.

Reply:
I meant to discuss with you. Do you think there is benefit of using `MPIR_Memcpy` as the the above comment suggests or universally use `MPIR_Typerep_pack` is sufficient? I am leaning toward the latter. If we want the `memcpy` shortcut, it should be optimized in `MPIR_Typerep_pack`.